test/testqsort.c

```diff
@@ -138,7 +130,7 @@ int main(int argc, char *argv[])
         test_sort("reverse sorted", nums, arraylen);
 
         for (i = 0; i < arraylen; i++) {
-            nums[i] = SDLTest_RandomInt(&rndctx);
+            nums[i] = SDL_rand_r(&seed, SDL_MAX_SINT32);
```

Comment:
While it doesn't break the test, this code sets a bad example:
* Passing SDL_MAX_SINT32 will give you a number in the range [0..0x8FFFFFFE].
* SDL_rand only gives roughly even odds for n up to about 1 million.

The correct way to get a full range SINT32 is `(Sint32)(SD_rand_bits_r(&seed) >>1)`. We could add a SDL_rand_sint32() function for this, but we decided against implementing the std::rand interface because people always use it wrong.

Reply:
It's fine for the test, no values out of range, but it's bad form. It generates lumpy random numbers, not uniform. :)
I mentioned it because it adds to the discussion of "most people use std::rand wrong".

Reply:
Fixed in the latest version, I use random numbers in the range of [0-1000000)